src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -81,6 +81,23 @@ public ElementNode(AbstractNode root, AbstractNode parent, DetailAST detailAst)
         createChildren();
     }
 
+    /**
+     * Compares current object with specified for order.
+     * @param other another {@code NodeInfo} object
+     * @return number representing order of current object to specified one
+     */
+    @Override
+    public int compareOrder(NodeInfo other) {
+        int result = 0;
+        if (other != null) {
+            result = getRootDepth(this) - getRootDepth(other);
+        }
+        if (result == 0 && other instanceof ElementNode) {
```

Comment:
Shouldn't this line be part of `if (other != null) {`?

Reply:
Spotbugs gave a warning for unchecked casting for that line, so to resolve that I put the check just before that, but I'll put them together now.

Reply:
done